kedro-datasets/kedro_datasets/partitions/partitioned_dataset.py

```diff
@@ -302,6 +307,12 @@ def load(self) -> dict[str, Callable[[], Any]]:
         return partitions
 
     def save(self, data: dict[str, Any]) -> None:
+        if self._use_async:
+            asyncio.run(self._async_save(data))
```

Comment:
If I understand correctly, `asyncio.run` creates a new event loop, so if there's already an event loop running (for example, in a Jupyter notebook), calling this will raise an error.

This is essentially the [red/blue function problem](https://journal.stuffwithstuff.com/2015/02/01/what-color-is-your-function/)... Most of Kedro is synchronous anyway AFAIK, but I think this might set an API expectation that could be difficult to satisfy cleanly.

@merelcht @ElenaKhaustova do you have more thoughts?

Reply:
> If I understand correctly, asyncio.run creates a new event loop, so if there's already an event loop running (for example, in a Jupyter notebook), calling this will raise an error.

That is indeed correct. Maybe it's alright though to say async saving doesn't work in interactive envs? 

> but I think this might set an API expectation that could be difficult to satisfy cleanly

@astrojuanlu can you elaborate what you mean with this?

Reply:
Sorry for the delay. What I meant is that use_async=True as it stands in this PR won't work (I think) in certain cases, and that fixing it might be hard